src/xrCore/command_line_key.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "xrCommon/xr_list.h"
+#include "_std_extensions.h"
+
+template < typename T > class XRCORE_API command_line_key;
+
+template < typename T >
+class XRCORE_API command_line_key
+{
+public:
+    command_line_key(pcstr flag_name, pcstr desc,T defval, bool req = false);
+    ~command_line_key();
+    bool IsProvided() const;          // was the option provided in the CLI?
+    T OptionValue() const;            // value provided with the option
+
+    static bool CheckArguments();
+    static void PrintHelp();
+    friend XRCORE_API bool ParseCommandLine(int argc, char* argv[]);
+
+private:
+    pstr option_name = nullptr;
+    pstr description = nullptr;
```

Comment:
There's absolutely no reason for these variables to be pstr. If there is, eliminate that reason.
```suggestion
    pcstr option_name = nullptr;
    pcstr description = nullptr;
```